js/map.js

```diff
@@ -0,0 +1,183 @@
+var pinListElement = document.querySelector('.map__pins');
+var cardListElement = document.querySelector('.map');
+var filters = document.querySelector('.map__filters-container');
+var photosfragment = document.createDocumentFragment();
+var fragment = document.createDocumentFragment();
+
+var avatars = [
+  'img/avatars/user01.png',
+  'img/avatars/user02.png',
+  'img/avatars/user03.png',
+  'img/avatars/user04.png',
+  'img/avatars/user05.png',
+  'img/avatars/user06.png',
+  'img/avatars/user07.png',
+  'img/avatars/user08.png'
+];
+
+var names = [
+  'Большая уютная квартира',
+  'Маленькая неуютная квартира',
+  'Огромный прекрасный дворец',
+  'Маленький ужасный дворец',
+  'Красивый гостевой домик',
+  'Некрасивый негостеприимный домик',
+  'Уютное бунгало далеко от моря',
+  'Неуютное бунгало по колено в воде'
+];
+
+var types = [
+  'palace',
+  'flat',
+  'house',
+  'bungalo'
+];
+
+var typesRus = {
+  'palace': 'Дворец',
+  'flat': 'Квартира',
+  'house': 'Дом',
+  'bungalo': 'Бунгало'
+};
+
+var times = [
+  '12:00',
+  '13:00',
+  '14:00'
+];
+
+var facilities = [
+  'wifi',
+  'dishwasher',
+  'parking',
+  'washer',
+  'elevator',
+  'conditioner'
+];
+
+var pictures = [
+  'http://o0.github.io/assets/images/tokyo/hotel1.jpg',
+  'http://o0.github.io/assets/images/tokyo/hotel2.jpg',
+  'http://o0.github.io/assets/images/tokyo/hotel3.jpg'
+];
+
+var getRandomInteger = function (min, max) {
+    return Math.floor(min + Math.random() * (max + 1 - min));
+};
+
+var getMixedArray = function (arr) {
+  for(var i = arr.length - 1; i > 0; i--){
+    var j = Math.floor(Math.random() * (i + 1));
+    var temp = arr[i];
+    arr[i] = arr[j];
+    arr[j] = temp;
+  }
+  return arr;
+};
+
+var getRandomСonsist = function (arr) {
+  getMixedArray(arr);
+  var qty = getRandomInteger(1, arr.length);
+  var consist = [];
+  for (var i = 0; i < qty; i++) {
+    consist.push(arr[i]);
+    }
+  return consist;
+}
+
+var getRandomElement = function (arr) {
+  return arr[Math.floor(Math.random() * arr.length)];
+};
+
+getMixedArray(avatars);
+
+var getElements = function (arr, arrLength) {
+  for (var i = 0; i < arrLength; i++) {
+    var obj = {};
+
+    obj.location = {
+      x: getRandomInteger(0, document.body.clientWidth),
+      y: getRandomInteger(130, 630)
+    };
+
+    obj.offer = {
+      title: getRandomElement(names),
+      address: obj.location.x + ' ,' + obj.location.y,
+      price: getRandomInteger(1000, 1000000),
+      type: getRandomElement(types),
+      rooms: getRandomInteger(1, 5),
+      guests: Math.floor(Math.random() * 10) + 1,
+      checkin: getRandomElement(times),
+      checkout: getRandomElement(times),
+      features: getRandomСonsist(facilities),
+      description: '',
+      photos: getMixedArray(pictures)
+    };
+
+    obj.author = {
+        avatar: avatars[i]
+    };
+
+   arr[i] = obj;
+  };
+  return arr;
+};
+
+var itemListLength = 8;
+var items = [];
+getElements(items, itemListLength);
```

Comment:
Make `getElements` works like this:
var items = getElements(itemListLength);
Your implementation looks weird for me...